### DISCUSSION TITLE
Create endpoint for promotion adjuster in OrderUpdater

### DISCUSSION THREAD

**Discussion by mamhoff** [solidusio/solidus#4487] [New Features or Ideas]
The code to handle promotions is scattered across four different classes: 
- Spree::OrderAdjuster
- Spree::OrderContents
- Spree::Promotion
- Spree::Adjustment (see the `#recalculate` method)

Move all the code related to applying a promotion including eligibility checks to a new configurable class, Spree::Promotion::OrderAdjuster.

**Comment by jarednorman** [comment]
You have a lot of ideas today.

This (like the others) seems like a good idea.
